media_driver/agnostic/common/codec/hal/codechal_utilities.h

```diff
@@ -252,6 +252,10 @@ MOS_STATUS CodecHalAllocateDataList(type **dataList, uint32_t length)
     {
         dataList[i] = &(ptr[i]);
     }
+    if(length <= 0)
```

Comment:
as length is uint32_t, <0 won't happen;
if ==0, no memory was alloced;
so this if() block is useless?